cf/regrid/utils.py

```diff
@@ -0,0 +1,1673 @@
+
+                    f.set_construct(
+                        f._DomainAncillary(source=value),
+                        key=key,
+                        axes=d_axes,
+                        copy=False,
+                    )
+
+
+def regrid_copy_coordinate_references(f, dst, dst_axis_keys):
+    """Copy coordinate references from the destination field to the new,
+    regridded field.
+
+    :Parameters:
+
+        f: `Field`
+            The source field.
+
+        dst: `Field`
+            The destination field.
+
+        dst_axis_keys: sequence of `str`
+            The keys of the regridding axes in the destination field.
+
+    :Returns:
+
+        `None`
+
+    """
+    dst_data_axes = dst.constructs.data_axes()
+
+    for ref in dst.coordinate_references(todict=True).values():
+        axes = set()
+        for key in ref.coordinates():
+            axes.update(dst_data_axes[key])
+
+        if axes and set(axes).issubset(dst_axis_keys):
+            # This coordinate reference's coordinates span the X
+            # and/or Y axes
+            f.set_coordinate_reference(ref, parent=dst, strict=True)
+
+
+def regrid_use_bounds(method):
+    """Returns whether to use the bounds or not in regridding. This is
+    only the case for conservative regridding.
+
+    :Parameters:
+
+        method: `str`
+            The regridding method
+
+    :Returns:
+
+        `bool`
+
+    """
+    return method in conservative_regridding_methods
+
+
+def regrid_update_coordinates(
+    f,
+    dst,
+    dst_dict,
+    dst_coords,
+    src_axis_keys,
+    dst_axis_keys,
+    cartesian=False,
+    dst_axis_sizes=None,
+    dst_coords_2D=False,
+    dst_coord_order=None,
+):
+    """Update the coordinates of the regridded field.
+
+    :Parameters:
+
+        f: `Field`
+            The regridded field. Updated in-place.
+
+        dst: Field or `dict`
+            The object containing the destination grid.
+
+        dst_dict: `bool`
+            Whether dst is a dictionary.
+
+        dst_coords: sequence
+            The destination coordinates.
+
+        src_axis_keys: sequence
+            The keys of the regridding axes in the source field.
+
+        dst_axis_keys: sequence
+            The keys of the regridding axes in the destination field.
+
+        cartesian: `bool`, optional
+            Whether regridding is Cartesian of spherical, False by
+            default.
+
+        dst_axis_sizes: sequence, optional
+            The sizes of the destination axes.
+
+        dst_coords_2D: `bool`, optional
+            Whether the destination coordinates are 2-d, currently only
+            applies to spherical regridding.
+
+        dst_coord_order: `list`, optional
+            A list of lists specifying the ordering of the axes for
+            each 2-d destination coordinate.
+
+    :Returns:
+
+        `None`
+
+    """
+    # NOTE: May be common ground between cartesian and shperical that
```

Comment:
(Not a typo introduced here, just moved I guess, not a functional one, but since I've seen it:)
```suggestion
    # NOTE: May be common ground between cartesian and spherical that
```